src/devices/GrovePi/Sensors/DhtSensor.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using Iot.Device.GrovePiDevice.Models;
+
+namespace Iot.Device.GrovePiDevice.Sensors
+{
+    /// <summary>
+    /// Type of DHT sensors
+    /// </summary>
+    public enum DhtType
+    {
+        Dht11 = 0,
+        Dht22 = 1,
+        Dht21 = 2,
+        Am2301 = 3
+    }
+
+    /// <summary>
+    /// DhtSensor supports DHT familly sensors
+    /// </summary>
+    public class DhtSensor : ISensor<double[]>
```

Comment:
could this perhaps share some code with https://github.com/dotnet/iot/tree/master/src/devices/Dhtxx ?

Reply:
At best, it can share the interface. The rest is not working the same way.
I think I'll open an issue to start a discussion on standard interface for various sensors. They will appear with different interface and can create confusion. We will have the same for few others.

Reply:
that's fine. what about moving this sensor outside of GrovePi directory so i.e. making it i.e.:
`src/devices/GroveDhtXXXX/`?

Reply:
I would leave it with GrovePi as it's really GrovePi specific. That said, if we fully reorganize the sensors, then why not. Still, this one is linked to GrovePi. The reading are done in GrovePi and there is a special function for that.